### ISSUE TITLE
Test failure: System.Net.Http.Functional.Tests.TelemetryTest_Http11.EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue

### ISSUE THREAD

**Issue by antonfirsov** [dotnet/runtime#40926]
Failed in #40841:
https://dev.azure.com/dnceng/public/_build/results?buildId=773535&view=ms.vss-test-web.build-test-results-tab&runId=24166882&paneView=debug&resultId=104442

Configurations:
```
net5.0-Windows_NT-Debug-x64-CoreCLR_release-Windows.10.Amd64.Server19H1.ES.Open
net5.0-Windows_NT-Release-x86-CoreCLR_release-Windows.10.Amd64.Server19H1.ES.Open
```

Stack Trace:
<details>

```
Child exception:
  Xunit.Sdk.DoesNotContainException: Assert.DoesNotContain() Failure
Found:    (filter expression)
In value: ConcurrentQueue<EventWrittenEventArgs> [EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = 1, EventName = "RequestStart", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, ...]
   at System.Net.Http.Functional.Tests.TelemetryTest_Http11.<>c.<<EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue>b__1_0>d.MoveNext() in /_/src/libraries/System.Net.Http/tests/FunctionalTests/TelemetryTest.cs:line 267
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 66

Child process:
  System.Net.Http.Functional.Tests, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Net.Http.Functional.Tests.TelemetryTest_Http11+<>c System.Threading.Tasks.Task <EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue>b__1_0()

```
</details>

**Comment by karelz** [comment]
Triage: Likely timing of new tests -- @MihaZupan can you please take a look?

**Comment by v-haren** [comment]
failed again in job: [runtime-libraries outerloop 20200827.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=789975&view=ms.vss-test-web.build-test-results-tab&runId=24927796&resultId=104058&paneView=debug)

failed test: 
System.Net.Http.Functional.Tests.TelemetryTest_Http11.EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue

Configurations:
net5.0-Windows_NT-Release-x86-CoreCLR_release-Windows.10.Amd64.Server19H1.Open


Error message
~~~
Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.


Stack trace

Child exception:
  Xunit.Sdk.DoesNotContainException: Assert.DoesNotContain() Failure
Found:    (filter expression)
In value: ConcurrentQueue<EventWrittenEventArgs> [EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = 1, EventName = "RequestStart", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, EventWrittenEventArgs { ActivityId = 00000000-0000-0000-0000-000000000000, Channel = None, EventId = -1, EventName = "EventCounters", EventSource = EventSource(System.Net.Http, d30b5633-7ef1-5485-b4e0-94979b102068), ... }, ...]
   at System.Net.Http.Functional.Tests.TelemetryTest_Http11.<>c.<<EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue>b__1_0>d.MoveNext() in /_/src/libraries/System.Net.Http/tests/FunctionalTests/TelemetryTest.cs:line 267
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 66

Child process:
  System.Net.Http.Functional.Tests, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Net.Http.Functional.Tests.TelemetryTest_Http11+<>c System.Threading.Tasks.Task <EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue>b__1_0()
~~~

**Comment by karelz** [comment]
Fixed in PR #41022
